src/main/java/org/springframework/data/elasticsearch/core/SearchHitSupport.java

```diff
@@ -81,4 +88,55 @@ public static Object unwrapSearchHits(Object result) {
 
 		return result;
 	}
+
+	public static <T> SearchPage<T> searchPageFor(SearchHits<T> searchHits, @Nullable Pageable pageable) {
```

Comment:
`toSearchPage()`, IMO sound better.

Reply:
For me, methods named _toSomething()_ transform the object that they are called upon into something different. Here we have a static utility function so I deliberately did not use _to.._